sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -174,7 +172,7 @@ def _claim_one_partition(self, ownership, **kwargs):
         try:
             self._update_ownership(new_ownership, **kwargs)
             return new_ownership
-        except (ResourceModifiedError, ResourceExistsError):
```

Comment:
we're removing the `ResourceModifiedError` exception handling and I feel it's related to the removed code
```
                etag=ownership["etag"],
                match_condition=MatchConditions.IfNotModified,
```

do you know under what circumstances `ResourceModifiedError` would be thrown?

Reply:
yeah, this error will be thrown when we pass `MatchConditions.IfNotModified` and an etag into `update_entity`, but the etag does not match/is not the most recently updated etag. So this will not occur if we don't pass in the `IfNotModified` condition (since it defaults to Unconditionally, which means it always updates and should not fail).

I'll add this back if I keep `IfNotModified`

Reply:
do we need to add `ResourceModifiedError` back here since we now introduce `IfNotModified` back?

Reply:
so I took a look at the JS implementation, and they don't raise an error, so I thought we shouldn't as well. https://github.com/Azure/azure-sdk-for-js/blob/7d82d3331de2dbd8db193f49bb783ae60c4a4fb9/sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts#L192

However, do we want to raise an error if the resource has been modified? Even though the behavior is different than JS?

Reply:
I'm thinking of the following case:

- 1 checkpointstore A on a node claimed a partition p_id
- 2 after a while, checkpointstore B on a different node found partition p_id is pickable (last modified time is long ago)
  - 2.1 maybe it's because the node that checkpointstore A is slow, it can't claim timely
- 3 checkpointstore B claimed partition p_id and modified the data
- 4 checkpointstore A is back trying to refresh the ownership of partition p_id and got ResourceModifiedError(?)

Isn't it the case that we should raise OwnerShipLostError? or my understanding of the ResourceModifiedError is not correct..

Reply:
See my comment above, I think the implementation in JS one is different, if there's a modified error in the JS code, they don't push the partition to an array, however, here the func we're implementing is to claim a single partition.

Reply:
yeah that makes sense that the error occurs b/c a different partition has already updated, so we would handle that with the OwnershipLostError. will add back in!